lib/crystalball/map_generator/parser_strategy/processor.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require 'parser/current'
+
+module Crystalball
+  class MapGenerator
+    class ParserStrategy
+      # Parses the given source files and adds the class and module definitions
+      # to the `consts_defined` array.
+      class Processor
+        attr_accessor :consts_defined, :consts_interacted_with, :current_scope
+
+        def on_send(node, _)
+          const = filtered_children(node).detect { |c| c.type == :const }
+          return unless const
+          add_constant_interacted(qualified_name_from_node(const), nil)
+        end
+
+        def on_casgn(node, scope)
+          namespace, name, = node.children
+          scope_name = namespace ? qualified_name(qualified_name_from_node(namespace), scope) : scope
+          add_constant_defined(name, scope_name)
+        end
+
+        def on_class(node, scope)
+          const, _, body = node.children # TODO: _ is superclass
+          process_class_or_module(const, body)
+        end
+
+        def on_module(node, scope)
+          const, body = node.children
+          process_class_or_module(const, body)
+        end
+
+        def consts_defined_in(path)
+          self.current_scope = nil
+          self.consts_defined = []
+          parse_and_process(path)
+          consts_defined
+        end
+
+        def consts_interacted_with_in(path)
+          self.current_scope = nil
+          self.consts_interacted_with = []
+          parse_and_process(path)
+          consts_interacted_with
+        end
```

Comment:
why all these handlers are public?

Reply:
You're right, they shouldn't be. They started like this because this class sublclassed `Parser::AST::Processor`, but that's no longer the case. I'll change it.